.github/workflows/docs-build.yaml

```diff
@@ -4,7 +4,11 @@ on:
   push:
     branches:
       - main
+    paths:
+      - 'docs/**'
```

Comment:
This would work if we only pushed docs from the *.rst files you see in the directory. Unfortunately, we also record our C++ API using exhale, therefore, if the documentation of our API (dictated by the comments of our C++ code) changes it won't be updated in our [gh-pages](https://github.com/pytorch/multipy/tree/gh-pages), and won't show up in our [docs](https://pytorch.org/multipy/latest/api/library_root.html).

The way our main docs get pushed is that once a new pr is merged, we use sphinx to generate docs in `docs/build`, we then push those docs to our gh-pages branch to be hosted [on our website under main](https://pytorch.org/multipy/main/). (You can test this out by modifying the rst files or modifying function comments in a file like `multipy/runtime/deploy.h`)

 Therefore, you need to check if anything changed after the docs have been built in `docs/build` (created when we run sphinx). This can be done with `git diff`, and exiting gracefully is described in [this thread](https://stackoverflow.com/questions/17797740/check-if-specific-file-in-git-repository-has-changed).